jamf_protect_api/scripts/python/generate_device_controls_report.py

```diff
@@ -0,0 +1,190 @@
+                    pageSize: $page_size
+                    next: $next
+                }
+            ) {
+                items {
+                        json
+                        eventType
+                        computer {
+                            hostName
+                        }
+                        created
+                    }
+                        pageInfo {
+                                next
+                    }
+                }
+            }
+        """
+
+
+def __main__():
+
+    if not all([PROTECT_INSTANCE, CLIENT_ID, PASSWORD]):
+        print("ERROR: Variables PROTECT_INSTANCE, CLIENT_ID, and PASSWORD must be set.")
+        sys.exit(1)
+
+    # Get the access token
+    access_token = get_access_token(PROTECT_INSTANCE, CLIENT_ID, PASSWORD)
+
+    cutoff_days = 30
+    cutoff_date = dt.datetime.now() - dt.timedelta(days=cutoff_days)
+
+    results = []
+    next_token = None
+    page_count = 1
+
+    print("Retrieving paginated results:")
+
+    while True:
+        print(f"  Retrieving page {page_count} of results...")
+        vars = {
+            "event_type": "auth-mount",
+            "created_cutoff": cutoff_date.isoformat() + "Z",
+            "page_size": 100,
+            "next": next_token,
+        }
+        resp = make_api_call(PROTECT_INSTANCE, access_token, LIST_ALERTS_QUERY, vars)
+        next_token = resp["data"]["listAlerts"]["pageInfo"]["next"]
+        results.extend(resp["data"]["listAlerts"]["items"])
+        if next_token is None:
+            break
+        page_count += 1
+    print(f"Found {len(results)} alerts matching filter.\n")
+
+    data_file = open(CSV_OUTPUT_FILE, "w", newline="")
+    csv_writer = csv.writer(data_file)
+    headers = [
+        "Timestamp",
+        "HostName",
+        "Serial",
+        "Vendor",
+        "Vendor ID",
+        "Product",
+        "Product ID",
+        "Device Serial",
+        "Encrypted",
+        "Action",
+    ]
+    csv_writer.writerow(headers)
+    for o in results:
+        raw_json = json.loads(o["json"])
+        hostname = raw_json["host"]["hostname"]
+        serial = raw_json["host"]["serial"]
+        vendorname = raw_json["match"]["event"]["device"]["vendorName"]
+        vendorid = raw_json["match"]["event"]["device"]["vendorId"]
+        productname = raw_json["match"]["event"]["device"]["productName"]
+        productid = raw_json["match"]["event"]["device"]["productId"]
+        devicesn = raw_json["match"]["event"]["device"]["serialNumber"]
+        isencrypted = raw_json["match"]["event"]["device"]["isEncrypted"]
+        action = (
+            raw_json["match"]["actions"][0]["name"]
+            + ", "
+            + raw_json["match"]["actions"][1]["name"]
+            + " & "
+            + raw_json["match"]["actions"][2]["name"]
+        )
+        time = raw_json["match"]["event"]["timestamp"]
+        timestamp = datetime.utcfromtimestamp(time).strftime("%Y-%m-%d %H:%M:%S")
+
+        rows = [
```

Comment:
Mixing up singular and plural in names causes confusion - this is defining a single row:
```suggestion
        row = [
```

Reply:
done